openglyph/include/openglyph/io/mega_file.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <khepri/io/exceptions.hpp>
+#include <khepri/io/file.hpp>
+#include <khepri/io/stream.hpp>
+
+#include <cstdint>
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace openglyph::io {
+
+/**
+ * @brief Represents a single MegaFile archive.
+ *
+ * This class provides functionality to open files stored inside
+ * a MegaFile archive (.meg). It manages the archive's metadata,
+ * including filenames and file info necessary for accessing
+ * individual files within the archive.
+ *
+ * Instances of this class are non-copyable and non-movable.
+ */
+class MegaFile final
+{
+public:
+    explicit MegaFile(const std::filesystem::path& mega_file_path);
+    ~MegaFile() = default;
+
+    MegaFile(const MegaFile&)                = delete;
+    MegaFile& operator=(const MegaFile&)     = delete;
+    MegaFile(MegaFile&&) noexcept            = delete;
+    MegaFile& operator=(MegaFile&&) noexcept = delete;
+
+    /**
+     * @brief Opens a file from the MegaFile archive.
+     *
+     * This method attempts to locate and open a file within the archive
+     * using the provided relative path. It first calculates the CRC32 hash
+     * of the path and uses it to look up the file's metadata in the archive.
+     * If the file is found, a `Stream` is returned to read its contents.
+     * Otherwise, a `nullptr` is returned.
+     *
+     * @param path The relative path of the file to open (as stored in the archive).
+     * @return A unique pointer to a `khepri::io::Stream` representing the file's
+     *         contents, or `nullptr` if the file was not found.
```

Comment:
Add some documentation about lifetimes: the returned Stream may not outlive the parent MegaFile.